src/coreclr/jit/lowerarmarch.cpp

```diff
@@ -48,13 +48,22 @@ bool Lowering::IsCallTargetInRange(void* addr)
 //    True if the immediate can be folded into an instruction,
 //    for example small enough and non-relocatable.
 //
-// TODO-CQ: we can contain a floating point 0.0 constant in a compare instruction
-// (vcmp on arm, fcmp on arm64).
-//
 bool Lowering::IsContainableImmed(GenTree* parentNode, GenTree* childNode) const
 {
     if (!varTypeIsFloating(parentNode->TypeGet()))
     {
+#ifdef TARGET_ARM64
+        if (parentNode->OperIsRelop() && childNode->IsCnsFltOrDbl() && childNode->IsFPZero())
```

Comment:
`IsFPZero()` already checks for `(gtOper == GT_CNS_DBL)`, so no need of `IsCnsFltOrDbl()` check.